src/com/google/enterprise/adaptor/prebuilt/PrebuiltTransforms.java

```diff
@@ -174,24 +228,53 @@ private CopyTransform(List<KeyPairing> copies, boolean overwrite,
     @Override
     public void transform(Metadata metadata, Map<String, String> params) {
       for (KeyPairing kp : copies) {
-        Set<String> values = metadata.getAllValues(kp.src);
+        Set<String> values = new TreeSet<String>();
+        switch (kp.src.keyset) {
+          case METADATA:
+            values.addAll(metadata.getAllValues(kp.src.key));
+            break;
+          case PARAMS:
+            if (params.get(kp.src.key) != null) {
+              values.add(params.get(kp.src.key));
+            }
+            break;
+        }
         if (values.isEmpty()) {
           log.log(Level.FINE, "No values for {0}. Skipping", kp.src);
           continue;
         }
         log.log(Level.FINE, "Copying values from {0} to {1}: {2}",
             new Object[] {kp.src, kp.dest, values});
-        Set<String> destValues = metadata.getAllValues(kp.dest);
-        if (!overwrite && !destValues.isEmpty()) {
-          values = new HashSet<String>(values);
-          log.log(Level.FINER, "Preexisting values for {0}. Combining: {1}",
-              new Object[] {kp.dest, destValues});
-          values.addAll(destValues);
+        switch (kp.dest.keyset) {
+          case METADATA:
+            Set<String> destValues = metadata.getAllValues(kp.dest.key);
+            if (!overwrite && !destValues.isEmpty()) {
+              log.log(Level.FINER, "Preexisting values for {0}. Combining: {1}",
+                  new Object[] {kp.dest, destValues});
+            values.addAll(destValues);
+            }
+            metadata.set(kp.dest.key, values);
+            break;
+          case PARAMS:
+            String value = values.iterator().next();
+            if (values.size() > 1) {
+              log.log(Level.FINER,
+                  "Multiple values for {0}. Using first value of {1} for {2}",
+                  new Object[] { kp.src, value, kp.dest });
+            }
+            params.put(kp.dest.key, value);
```

Comment:
OK, my preference is for the default value of overwrite to be false for metadata, but true for params. I think that produces the least surprising results, but still lets you explicitly set overwrite=false if you want to not change a previously set value.

I was thinking that the constructor argument and field could be a Boolean: TRUE, FALSE, or null, where null is interpreted as FALSE by dest.keyset == METADATA, and TRUE by dest.keyset == PARAMS.